oxide_test.go

```diff
@@ -6,79 +6,95 @@
+		opts = defaultMockDriverOptions()
+	})
+
+	Describe("SetConfigFromFlags", func() {
+		It("should succeed when all required fields are given", func() {
+			Expect(SUT.SetConfigFromFlags(opts)).To(Succeed())
+		})
+
+		Describe("errors", func() {
+			DescribeTable("should fail when a required string field is missing",
+				func(fields []string) {
+					for _, field := range fields {
+						opts.Data[field] = ""
+					}
+					err := SUT.SetConfigFromFlags(opts)
+					Expect(err).To(HaveOccurred())
+				},
+				Entry("host", []string{flagHost}),
+				Entry("token", []string{flagToken}),
+				Entry("project", []string{flagProject}),
+				Entry("diskImageId", []string{flagBootDiskImageID}),
+			)
+
+			It("should fail when nothing is given", func() {
+				err := SUT.SetConfigFromFlags(&commandstest.FakeFlagger{
+					Data: map[string]interface{}{},
+				})
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("required option not set: oxide-host"))
+				Expect(err.Error()).To(ContainSubstring("required option not set: oxide-token"))
+				Expect(err.Error()).To(ContainSubstring("required option not set: oxide-project"))
+				Expect(err.Error()).To(ContainSubstring("required option not set: oxide-boot-disk-image-id"))
+			})
+		})
+	})
+
+	DescribeTable("RancherMachineState mapping is correct",
+		func(instanceState oxide.InstanceState, expectedState state.State) {
+			Expect(toRancherMachineState(instanceState)).To(Equal(expectedState))
 		},
-		{
-			name:          "unknown",
-			instanceState: oxide.InstanceState("unknown"),
-			expectedState: state.None,
+		Entry("creating", oxide.InstanceStateCreating, state.Starting),
+		Entry("starting", oxide.InstanceStateStarting, state.Starting),
+		Entry("running", oxide.InstanceStateRunning, state.Running),
+		Entry("stopping", oxide.InstanceStateStopping, state.Stopping),
+		Entry("stopped", oxide.InstanceStateStopped, state.Stopped),
+		Entry("repairing", oxide.InstanceStateRepairing, state.Stopped),
+		Entry("rebooting", oxide.InstanceStateRebooting, state.Stopping),
+		Entry("migrating", oxide.InstanceStateMigrating, state.Running),
+		Entry("failed", oxide.InstanceStateFailed, state.Error),
+		Entry("destroyed", oxide.InstanceStateDestroyed, state.Error),
+		Entry("unknown", oxide.InstanceState("unknown"), state.None),
+	)
+})
+
+func defaultMockDriverOptions() (rv *commandstest.FakeFlagger) {
+	rv = &commandstest.FakeFlagger{
+		Data: map[string]interface{}{
+			//flagHost:                      "",
+			//flagToken:                     "",
+			//flagProject:                   "",
+			//flagVCPUs:                     1,
+			//flagMemory:                    1024,
+			//flagBootDiskSize:              20,
+			//flagBootDiskImageID:           "",
+			//flagVPC:                       "",
+			//flagSubnet:                    "",
+			//flagUserDataFile:              "",
+			//flagSSHUser:                   "",
+			//flagAdditionalSSHPublicKeyIDs: []string{},
```

Comment:
What's the purpose for these commented out items?

Reply:
removing those.. not needed as I thought